sunpy/net/dataretriever/sources/stereo.py

```diff
@@ -48,16 +49,17 @@ def _get_url_for_timerange(self, timerange, **kwargs):
         """
         returns list of urls corresponding to given TimeRange.
         """
+        if not timerange:
+            return []
         regex = re.compile('[^a-zA-Z0-9]')
-        source = kwargs.get('source').lower() 
+        source = kwargs.get('source')
+        table_ = {'STEREO_A':'ahead', 'STEREO_B':'behind'} 
         detector = regex.sub('', kwargs.get('detector')).lower() #default to euvi ?
         table = { 'euvi':'euvi', 'hi1' : 'hi_1', 'hi2': 'hi_2', 'cor2':'cor2'}
```

Comment:
I would have to include three ```if``` statements for ```hi1``` , ```hi2``` and the rest if I don't use ```table```. This looks much better, I would add some comments to make its utility clearer.